Detectors/GlobalTrackingWorkflow/tofworkflow/src/RecoWorkflowSpec.cxx

```diff
@@ -103,24 +91,19 @@ class TOFDPLRecoWorkflowTask
 
     // we do a copy of the input but we are looking for a way to avoid it (current problem in conversion form unique_ptr to *)
 
-    auto tracksRO = std::make_shared<std::vector<o2::dataformats::TrackTPCITS>>();
-    //  std::vector<o2::dataformats::TrackTPCITS> tracksRO;
-    *tracksRO.get() = std::move(*tracks);
-    // for (int i = 0; i < tracks->size(); i++) {
-    //   tracksRO.emplace_back(tracks->at(i));
-    // }
-    auto clustersRO = std::make_shared<std::vector<o2::tof::Cluster>>();
-    //std::vector<o2::tof::Cluster> clustersRO;
-    *clustersRO.get() = std::move(*clusters);
-    //  for (int i = 0; i < clusters->size(); i++) {
-    //    clustersRO.emplace_back(clusters->at(i));
-    // }
-
-    if (mUseMC)
-      mMatcher.initWorkflow(tracksRO.get(), clustersRO.get(), &toflab, itslab.get(), tpclab.get());
-    else
-      mMatcher.initWorkflow(tracksRO.get(), clustersRO.get(), nullptr, nullptr, nullptr);
+    gsl::span<const o2::MCCompLabel> itslab;
+    gsl::span<const o2::MCCompLabel> tpclab;
+    o2::dataformats::MCTruthContainer<o2::MCCompLabel> toflab;
+    if (mUseMC) {
+      const auto toflabel = pc.inputs().get<o2::dataformats::MCTruthContainer<o2::MCCompLabel>*>("tofclusterlabel");
+      itslab = pc.inputs().get<gsl::span<o2::MCCompLabel>>("itstracklabel");
+      tpclab = pc.inputs().get<gsl::span<o2::MCCompLabel>>("tpctracklabel");
+      toflab = std::move(*toflabel);
 
+      mMatcher.initWorkflow(&tracksRO, &clustersRO, &toflab, &itslab, &tpclab);
+    } else {
+      mMatcher.initWorkflow(&tracksRO, &clustersRO, nullptr, nullptr, nullptr);
```

Comment:
no, please change the initWorkflow to take as an argument to span objects or references, not the pointers!
And remove here this else..., just pass empty spans if the MC is not requested

Reply:
Hi Ruben,
just for my curiosity. What is the reason for having references instead of pointers?
Cheers